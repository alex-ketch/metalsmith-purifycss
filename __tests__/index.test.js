const Metalsmith = require('metalsmith');
const purifyCSS = require('../lib/index');

const build = options => {
  return new Promise ((resolve, reject) => {
    Metalsmith(`${__dirname }/fixtures`)
    .source('src')
    .use(purifyCSS(options))
    .build((err, files) => {
      if (err) reject(err);
      resolve(files);
    });
  })
};

test('purified CSS file is created', () => {
  const options = {
    content: ['*.html', '*.js'],
    css: ['styles.css'],
    output: 'styles-cleaned.css',
  };

  return build(options).then((files) => {
    expect(files['styles-cleaned.css']).toBeDefined();
  })
});

test('original CSS is purified', () => {
  const options = {
    content: ['*.html', '*.js'],
    css: ['styles.css'],
    output: 'styles-cleaned.css',
  };

  return build(options).then((files) => {
    expect(files['styles-cleaned.css'].contents.toString()).not.toEqual(files['styles.css']);
  })
});

test('unused selectors are removed', () => {
  const options = {
    content: ['*.html', '*.js'],
    css: ['styles.css'],
    output: 'styles-cleaned.css',
  };

  return build(options).then((files) => {
    expect(files['styles-cleaned.css'].contents.toString()).toMatchSnapshot();
  })
});

test('output is minified', () => {
  const options = {
    content: ['*.html', '*.js'],
    css: ['styles.css'],
    output: 'styles-cleaned.css',
    minify: true
  };

  return build(options).then((files) => {
    expect(files['styles-cleaned.css'].contents.toString()).toMatchSnapshot();
  })
});

test('whitelist option is respected', () => {
  const options = {
    content: ['*.html', '*.js'],
    css: ['styles.css'],
    output: 'styles-cleaned.css',
    whitelist: ['#whitelisted']
  };

  return build(options).then((files) => {
    expect(files['styles-cleaned.css'].contents.toString()).toContain('#whitelisted');
  })
});

test('an array of stylesheets is accepted', () => {
  const options = {
    content: ['*.html', '*.js'],
    css: ['styles.css', 'styles2.css'],
    output: 'styles-cleaned.css',
  };

  return build(options).then((files) => {
    expect(files['styles-cleaned.css'].contents.toString()).toContain('.multipleCSSfiles');
  })
});


test('a glob is accepted for CSS parameter', () => {
  const options = {
    content: ['*.html', '*.js'],
    css: ['*.css'],
    output: 'styles-cleaned.css',
  };

  return build(options).then((files) => {
    expect(files['styles-cleaned.css'].contents.toString()).toContain('span');
  })
});
