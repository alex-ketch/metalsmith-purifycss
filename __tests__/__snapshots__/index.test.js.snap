// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`output is minified 1`] = `".unusual-JS\\\\(class\\\\),.unusual\\\\(class\\\\){color:#00f}.button-active,.button-react{border:2px solid red}.className{font-style:italic}.nestedFiles{border:2px solid #000}"`;

exports[`unused selectors are removed 1`] = `
".className {
  font-style: italic;
}

.unusual\\\\(class\\\\) {
  color: blue;
}

/* JavaScript */

.button-active {
  border: 2px solid red;
}

.unusual-JS\\\\(class\\\\) {
  color: blue;
}

.button-react {
  border: 2px solid red;
}

.nestedFiles {
  border: 2px solid black;
}"
`;
